src/auth/oauth2client.ts

```diff
@@ -749,7 +754,13 @@ export class OAuth2Client extends AuthClient {
       !this.credentials.access_token || this.isTokenExpiring();
     if (shouldRefresh) {
       if (!this.credentials.refresh_token) {
-        throw new Error('No refresh token is set.');
+        const downscopedCreds = await this.refreshHandler();
+        if (downscopedCreds) {
```

Comment:
Based on the above, this is not sufficient as `downscopedCreds.access_token` may be null.
I suggest requiring these fields.

Reply:
ok, I add a check for all 3 cases after calling refreshHandler and but calling setCredentials.
But would this be redundant with the above comment that throw error if refreshHandler returns no token 
being returned.